api_definition/coepi_0.2.0.yml

```diff
@@ -0,0 +1,144 @@
+openapi: 3.0.2
+info:
+  description: This is the API model for CoEpi Server
+  version: 0.2.0
+  title: CoEpi Server
+  license:
+    name: MIT License
+    url: https://github.com/Co-Epi/data-models/blob/master/LICENSE
+    
+servers:
+  - url: https://coepi.wolk.com:8080
+    description: URL Endpoint used for dev/test
+    
+paths:
+  /cenreport:
+    post:
+      summary: Submit symptom or infection report
+      description: Users submit symptom / infection reports and reveal the secret CENKey in their application
+      
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                reportId: # Suggestion: Why is this being passed by the client? Better to be auto-generated by server?
+                  type: string
+                  description: Unique Identifier of the infection report
+                report:
+                  type: string
+                  description: Content of the infection report. Example, symptoms
+                cenKeys:
+                  type: array
+                  items:
+                    $ref: '#/components/schemas/CenKey'
+                  minItems: 1
+                  maxItems: 10 # TODO: Discuss appropriate value
+                  uniqueItems: true 
+              required:
+                - reportId
+                - report
+                - cenKeys
+                
+      responses:
+        '200':
+          description: The report was submitted successfully
+        '400':
+          description: Request Parameter Invalid # TBD
+        '500':
+          description: Internal Server Error
+        default:
+          description: Unexpected Error
+  
+  
+  /cenreport/{cenkey}: # Suggestion: Should we allow passing a list of CEN keys to reduce roundtrips ?
+    get:
+      summary: Returns a list of reports associated with a CEN Key
+      parameters:
+      - in: path
+        name: cenkey
+        description: CEN Key for which we need to query the reports
+        required: true
+        schema:
+          $ref: '#/components/schemas/CenKey'
+      
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Report'
```

Comment:
If this is made batch, would `Report` need a circular reference to the CEN it was from? Or would this need to become a map of CEN -> Report(s)?

Reply:
The latter approach sounds cleaner.